src/http/client.js

```diff
@@ -30,6 +31,35 @@ export default class HttpClient {
 		}
 	}
 
+	/**
+	 * Perform an http request using the given method
+	 * @param {String} method
+	 * @param {String} url
+	 * @param {Object} headers
+	 * @param {Object} body
+	 * @return {Object}
+	 */
+	async request(method, url, headers, body) {
+		const contentType =
+			body instanceof FormData
+				? 'multipart/form-data'
+				: 'application/json';
+		const response = await fetch(url, {
+			method,
+			headers: Object.assign(headers, {'Content-Type': contentType}),
```

Comment:
mmm maybe you should use one inmutable library instead of so many object.assign

Reply:
I only use this to copy the contents.

I used to have:

```
```

Reply:
But It needs an extra babel plugin and Liferay's eslint configuration doesn't like it

Reply:
d'oh. I prefer the ... notation. But you are just doing it becase you want a one liner, right?

If you don't want inmutable objects nor one-liners, headers['Content-Type'] = contentType is  clearer.